core/src/main/java/io/questdb/cairo/TableReader.java

```diff
@@ -103,7 +102,13 @@ public TableReader(CairoConfiguration configuration, CharSequence tableName, @Nu
                     .I$();
             this.txFile = new TxReader(ff).ofRO(path, partitionBy);
             path.trimTo(rootLen);
-            readTxnSlow();
+
+            long deadline = this.configuration.getMicrosecondClock().getTicks() + this.configuration.getSpinLockTimeoutUs();
+            long prevStructVersion = metadata.getStructureVersion();
+            do {
+                readTxnSlow(deadline);
+            } while (!reloadStruct(prevStructVersion, txFile.getStructureVersion(), deadline));
```

Comment:
`reloadStuct()` is a waiting loop itself, expecting version. I would expect this to be a single loop, which would reload both _txn and _meta in the same iteration. Otherwise there is potential timeout if metadata never comes, which might happen

Reply:
Deadline if fixed for all waits, if meta load times out then it's a game over. `reloadStruct` will not re-try if structure version does not match it will return false. I cannot see how to write it in 1 loop, each step may need a retry